src/utils/self_test/self_test.c

```diff
@@ -42,25 +48,124 @@ struct st_master_endpt {
 	int32_t test_completed;
 };
 
-static const char * const crt_st_msg_type_str[] = { "EMPTY",
-						    "IOV",
-						    "BULK_PUT",
-						    "BULK_GET" };
+const struct {
+	char identifier;
+	char *short_name;
+	char *long_name;
+	enum crt_st_msg_type type;
+} transfer_type_map[] = {{'e', "E", "EMPTY", CRT_SELF_TEST_MSG_TYPE_EMPTY},
+			 {'i', "I", "IOV", CRT_SELF_TEST_MSG_TYPE_IOV},
+			 {'b', "Bp", "BULK_PUT",
+			  CRT_SELF_TEST_MSG_TYPE_BULK_PUT},
+			 {'r', "Bg", "BULK_GET",
+			  CRT_SELF_TEST_MSG_TYPE_BULK_GET} };
+
+#define MSG_TYPE_STR(str, id)					\
+	{	\
+	int _i;	\
+	for (_i = 0; _i < 4; _i++) {				\
+		if (id == transfer_type_map[_i].type) {		\
+			str =  transfer_type_map[_i].long_name;	\
+			break;					\
+		}						\
+	}	\
+	}
+
+#define	TST_HIGH	0x01	/* test in higher direction */
+#define	TST_LOW		0x02	/* test in lower direction */
+#define TST_OUTPUT	0x10	/* print results */
+typedef struct {
```

Comment:
(style)  do not add new typedefs

Reply:
This is a typedef to keep properties for each stats together rather than having multiple arrays, one for each property, and then trying to make them consistent.  The latter is error prone.

You need to update rules for typedefs.  There are correct and incorrect usage of them and you shouldn't make a blanket statement.